### PR TITLE
Use formik in SendAssetsModal

### LINKED ISSUE #1891: Terrible keyboard behavior on mobile
https://github.com/decentdao/decent-interface/assets/62394931/9f829183-4782-4d81-8765-0cd562e7bda6

Really poor performance typing in address for send assets on mobile. Keyboard disappears after every key stroke, and the UI is generally very jumpy.

The question mark icon doesn't do anything on tap or press.

_Originally posted by @nicolaus-sherrill in https://github.com/decentdao/decent-interface/discussions/1863#discussioncomment-9524860_

### PR DESCRIPTION
**Pull request by DarksightKellar**
Closes #1891 

This PR replaces the deprecated use of `EthAddressInput` with `AddressInput` and Formik.
This fixes the bug causing the keyboard to be dismissed on each keystroke.

Looking for feedback on having the entire "form" controlled by Formik. Currently it's just the address input field, even if the whole "form" is wrapped in Formik and Form.

### CODE REVIEW

**Comment by adamgall** [comment]
@DarksightKellar what do you mean by "having the entire "form" controlled by Formik"? Are you referring to things like validation? Anything else?

**Comment by DarksightKellar** [comment]
> @DarksightKellar what do you mean by "having the entire "form" controlled by Formik"? Are you referring to things like validation? Anything else?

@adamgall  Yeah, validation and value updates (which are currently being set and read from component-level state, instead of via form's `values`)

**Comment by adamgall** [comment]
> > @DarksightKellar what do you mean by "having the entire "form" controlled by Formik"? Are you referring to things like validation? Anything else?
> 
> @adamgall Yeah, validation and value updates (which are currently being set and read from component-level state, instead of via form's `values`)

I assume you mention it in the comments of this PR because you think it would be "more correct" to use Formik for these validation and value updates?

How much more work will it take to implement and test that?

**Review by Da-Colon — COMMENTED** [review]
pending your look into updating functionality to formik, this does resolve the issue.

**Review by tomstuart123 — APPROVED** [review]
this solves the problem for me on mobile

File: src/components/ui/modals/SendAssetsModal.tsx [hunk 1]
```diff
@@ -47,110 +44,150 @@ export function SendAssetsModal({ close }: { close: () => void }) {
...
+
+  const submitSendAssets = async (values: { destinationAddress: string }) => {
+    const { destinationAddress } = values;
+
+    await sendAssets({
+      transferAmount: inputAmount?.bigintValue || 0n,
+      asset: selectedAsset,
+      destinationAddress,
+      nonce: nonceInput,
+      submitProposal,
```
[the remark this answers is not in the document]
Comment (DarksightKellar) [reply src/components/ui/modals/SendAssetsModal.tsx:64]: I hear you, but going back to `useSendAssets` (as it was previously) doesn't jive with the way Formik handles its form field values, which seem to be all localised only to the form. 

There is `useFields`, which I've tried to use, but seems would require quite a bit a refactoring for it to work (see [docs here](https://formik.org/docs/api/useField)).

I personally had a cryptic error when I tried to incorporate it in such a way as to maintain the current logic: "Cannot read properties of undefined (reading 'getFieldProps')".

Gonna leave as is for now.

**Pull request merged**